doc/ztp/inband_ztp.md

```diff
@@ -0,0 +1,246 @@
+```
+{
+	"ztp": {
+		"01-configdb-json": {
+			"url": {
+				"source": "http://192.168.0.1/ZTP/config_db.json",
+				"destination": "/etc/sonic/config_db.json"
+			}
+		},
+		"02-configdb-json": {
+			"url": {
+				"source": "http://192.168.0.1/ZTP/config_db-FEATURE.json",
+				"destination": "/etc/sonic/config_db.json"
+			},
+			"dynamic-url": {
+				"source": {
+					"prefix": "http://192.168.0.1/ZTP/",
+					"identifier": "hostname",
+					"suffix": "_config_db.json"
+				},
+				"destination": "/etc/sonic/config_db.json"
+			}
+		},
+		"03-script": {
+			"plugin": {
+				"url": {
+					"source": "http://192.168.0.1/ZTP/03-script.sh"
+				}
+			},
+			"reboot-on-failure": "true"
+		}
+	}
+}
+```
+- Last step is to clear ZTP configuration from the switch:
+1. Clear ZTP and COPP tables from config DB using sonic-db-cli.
+2. Clear ZTP and COPP tables from config_db.json.
+3. Delete file ztp_dhcp.json and restart interfaces-config service. When ztp_dhcp.json does not exist, interfaces-config service will create dhclient.conf without request of ZTP options.
+
+### 3.1.4 SWSS COPP
+For making DHCP packets arrive to CPU, need to register the following traps:
+- SAI_HOSTIF_TRAP_TYPE_DHCP_L2
+- SAI_HOSTIF_TRAP_TYPE_DHCPV6_L2
+
+Service config-setup creates new config_db.json during init and performs config relaod to load it. The following will be added to this config_db.json:
+```
+COPP_TRAP|l2dhcp
+  always_enabled:true
+```
+ZTP engine will delete this key from config DB when ZTP finishes its work.
+
+COPP manager listens to changes in COPP_TRAP table, therefore will handle and register the traps.
+When COPP manger handles changes in COPP_TRAP table it uses data being read in init and kept in the cache, this data is generated from copp_cfg.j2.
+
+Following will be added to copp_cfg.j2:
+```
+"l2dhcp": {
+  "trap_ids": "l2dhcp,l2dhcpv6",
+  "trap_group": "queue4_group4"
+},
+```
+```
+"queue4_group5": {
+  "trap_action":"copy",
+  "trap_priority":"4",
+  "queue": "4",
+  "meter_type":"packets",
+  "mode":"sr_tcm",
+  "cir":"600",
+  "cbs":"600",
+  "red_action":"drop"
+}
+```
+And extend trap_id_map with:
+```
+static map<string, sai_hostif_trap_type_t> trap_id_map = {
+  {"l2dhcp", SAI_HOSTIF_TRAP_TYPE_DHCP_L2},
+  {"l2dhcpv6", SAI_HOSTIF_TRAP_TYPE_DHCPV6_L2}
+};
+```
+
+## 3.2 ZTP provision over in-band network on runtime
+User can force ZTP service to run using CLI command "ztp run", this will restart ZTP operation.
+In this scenario, ztp-engine performs the following:
+1. Verify all ZTP sub features (inband, ipv4, ipv6) are enabled in config DB. If not, create config_db.json and ztp_dhcp.json (same as config-setup does on init) and perfrom config reload to load the newly created config_db.json.
+2. Rest of the flow is as described in [3.1.3 ZTP service](#313-ztp-service)
+
+## 3.3 Config DB
+When ZTP is enabled, the following will be added to config DB at init and will be deleted once ZTP will finish its operation.
+```
```

Comment:
How to check the config below? from redis or config_db.json?

Reply:
Both methods are valid.